### ISSUE TITLE
Refactor bootstrap.php

### ISSUE THREAD

**Issue by maddes** [maddes/carrito#29]
There's a lot of code that should go in the constructor of each class.

It would be great to do a wacky mapper for the IoC so we don't have so many wasted lines on stuff like:

```
$registry->set('affiliate', new Affiliate($registry));
```

Also, isn't that a circular reference?

**Comment by maddes** [comment]
A mapper property can be added to the `Registry` class.
Then, in the get function can be made something like:

``` php
        if (array_key_exists($key, $this->data))
        {
            return $this->data[$key];
        }
        elseif (array_key_exists($key, $this->mapper))
        {
            return $this->data[$key] = new $this->mapper[$key]($this);
        }
        else
        {
            return null;
        }
```

**Comment by maddes** [comment]
This generates issues where dependencies are assumed (actually, that's a good thing).
For example, accessing `$_SERVER` directly without loading the `Request` class.

**Comment by maddes** [comment]
At least 61 illegal usages of $_SERVER found with this…

**Comment by maddes** [comment]
Solved in v4.0.0
